src/pages/AddItem.js

```diff
@@ -1,10 +1,30 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { db } from '../lib/firebase';
+import swal from 'sweetalert';
+import { useHistory } from 'react-router-dom';
 
 export default function AddItem(props) {
   const [itemName, setItemName] = useState('');
   const [purchaseFrequency, setPurchaseFrequency] = useState(null);
   const [lastPurchased, setLastPurchased] = useState(null);
+  const [listItems, setListItems] = useState([]);
+
+  const history = useHistory();
+
+  useEffect(() => {
+    // do we need a check for loading if grabbing firestore data takes too long
+    // possibly use the 'loading' property from useCollection
```

Comment:
@jamesncox @jssckbl Ooo yes - I really like the idea of using the loading property from `useCollection`. I haven't used that hook myself, but the [documentation](https://github.com/csfrequency/react-firebase-hooks/tree/3f04ec5669d2a9ff898818792d99b384a00eb3f8/firestore#usecollection) looks pretty thorough and I think you could adapt your code without many changes! You could also build your own loading property using `useState`, but using built-in properties/methods sounds great!

You might also consider moving this function to a `useEffect` in `App.js` and then passing it down to `AddItem.js` and `List.js` so that you aren't subscribing/unsubscribing frequently and unnecessarily when navigating between pages.

Reply:
@laurenyz we did indeed refactor our fetch to the Firestore database, pulling it out of `AddItem.js` and placing it inside `App.js`.

The important updates to our `useEffect()` are adding `token` to the dependency array and an `if (token)` conditional to allow the `useEffect()` to run. (If you initialize the app without the `if (token)` conditional, and there is no token yet, you get an error from firestore not accepting an empty string as a parameter inside `db.collection(token)`.

From there we pass down the `listItems`, `setListItems` and `token` props to `AddItem.js`

Reply:
Nice! Sounds great!